src/main/java/io/r2dbc/h2/codecs/DefaultCodecs.java

```diff
@@ -114,4 +115,16 @@ public Value encodeNull(Class<?> type) {
 
         return null;
     }
+
+    /**
+     * Adds secondary codecs based on different conditions, e.g. Classpath availability.
+     * @return this {@link DefaultCodecs}
+     */
+    public DefaultCodecs addSecondaryCodecs() {
+        if (Check.findClass("org.locationtech.jts.geom.Geometry")) {
+            this.codecs.add(new GeometryCodec());
+        }
+
+        return this;
+    }
```

Comment:
I'm not convinced we need a public method for this. I think simply adding the extra check after creating the primary list is sufficient.

We can add this API in the future if extensibility is needed.

Reply:
Yes, let's move this functionality into the constructor for the time being.

Reply:
Summarizing required change:
* Move `addSecondaryCodecs` call to the end of  `DefaultCodecs`'s  constructor . (Keeping `H2ConnectionFactory` intact )

Ok?